lib/ansible/parsing/vault/__init__.py

```diff
@@ -1139,50 +1074,101 @@ def _editor_shell_command(self, filename):
 ########################################
 #               CIPHERS                #
 ########################################
+class VaultCipher:
+
+    @classmethod
+    @abstractmethod
+    def encrypt(cls, vault, secret):
+        pass
+
+    @classmethod
+    @abstractmethod
+    def decrypt(cls, vault, secret):
+        pass
+
+    @staticmethod
+    def _get_salt():
+        custom_salt, origin = C.config.get_config_value_and_origin('VAULT_ENCRYPT_SALT')
+        if custom_salt:
+            if not origin.startswith('cli:'):
+                display.warning('Using configured salt, reusing a salt across vaults is insecure')
+        else:
+            # Future: replace with call to ansible.utils.encrypt.random_salt(32)
+            custom_salt = os.urandom(32)
+        return to_bytes(custom_salt)
+
+    @staticmethod
+    def _is_equal(b_a, b_b):
```

Comment:
Unless I've overlooked something, this function is unused, and has been for some time (outside of unit tests).

Reply:
i just moved the existing one, i did not check for usage in current nor any of the backport targets

Reply:
last used in 2.11 (outside tests)
```
lib/ansible/parsing/vault/__init__.py:        if not cls._is_equal(b_crypted_hmac, to_bytes(hmac_decrypt.hexdigest())):
```